src/launcher.c

```diff
@@ -406,6 +446,11 @@ static int load_cxlib(char *cxlib_args)
     }
   }
 
+  if (mprotect(p, sh_size, PROT_READ | PROT_WRITE) != 0) {
+    warn("mprotect(PROT_READ | PROT_WRITE)");
```

Comment:
Maybe be more explicit in the warning or add a comment about what this code does